docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+
+Most of the time you will receive feedback from the review. Please be prepared to provide more information or details
+and update your submission, if required.
+
+Common aspects that a review looks for:
+
+1. Are there any defects and is the submission ready to be widely distributed?
+2. Does the submission provide real additional value to the project that will improve what users will get out of the
+   software?
+3. Does the submission include automated tests (e.g. unit test) when applicable that will ensure that the implementation
+   does what it
+   is intended to do?
+4. Is the copyright of the submission clear, compatible with the project and non-gratuitous?
+5. Commits well formatted, cover a single topic and independent?
+6. Is the documentation updated to reflect the changes?
+7. Does the implementation follow the general style of the project?
+
+## Format of Commit Messages
+
+Each commit should be similar to the following format to help with the review process:
+
+```
+<Module>: Capitalized, short summary (the 'what' of the change)
+
+Optional, more detailed explanation of what the commit does (the 'why' and 'how').
+
+Signed-off-by: My Name <myemail@example.org>
+```
+
+`Module` should be a directory name, filename or similar to help understand what part of the application is affected
+by the change.
+
+It is important to have a "Signed-off-by" line on each commit to certify that you agree to the
+[developer certificate of origin](developer-certificate-of-origin.md). Depending on your IDE or editor you can
+automatically add this submission line with each commit.
+It has to contain your real name (please don't use pseudonyms) and contain a current email address. Unfortunately, we
+cannot accept anonymous submissions.
+
+## Format of the Pull Request
+
+The pull request title and description will help the contributors to describe the change that is finally merged into
+the main branch. Each submission will be squashed into a single commit before the merge.
+
+This project follows the [Conventional Commits specification](https://www.conventionalcommits.org) to help to easily
+understand the intention and change of a commit.
+
+When crating the pull request we ask you to use the following guideline:
+
+The title of the pull request has the following format:
+
+```
+<type>: Capitalized, short summary (the 'what' of the change)
+```
+
+The <type> may be one of the following list:
+
+* `feat` - A new feature
+* `fix` - A bug fix
+* `test` - Adding a new test or improve an existing one
+* `docs` - Changes or additions to the documentation
+* `refactor` - Refactoring of the code or other project elements
+* `chore` - Other modifications that do not modify implementation, test or documentations
+
+If your submission does introduce a breaking change please add `BREAKING CHANGE` to the beginning of the description.
+
+Similar to a commit the description describes the overall change of the submission and include a `Signed-off-by` line
```

Comment:
Could help to mention `git commit -s` and also usage of `verified`label which requires a GPG Key registered along with the according git account

Reply:
`verified` would mean to enforce signed commits. Do you think this should be enforced?